From 21b5e0225ed4f41887ebaf986a87615cde156c14 Mon Sep 17 00:00:00 2001
From: Julien Thomas <jthomas@zenetys.com>
Date: Wed, 10 Aug 2022 17:39:58 +0200
Subject: [PATCH] build: Fix #pragma GCC diagnostic not allowed inside
 functions

Commit 829a6f6 "CI: improve use of github action matrix feature;
optimization" has introduced use of #pragma GCC diagnostic in function
doNameLine() of runtime/conf.c. This is not supported by old compilers
like GCC 4.4.7 on el6.
---
 runtime/rsyslog.h | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/runtime/rsyslog.h b/runtime/rsyslog.h
index 3f7cfc62c..fa7280ae8 100644
--- a/runtime/rsyslog.h
+++ b/runtime/rsyslog.h
@@ -110,8 +110,15 @@
 						_Pragma("GCC diagnostic ignored \"-Wformat-nonliteral\"")
 	#define PRAGMA_IGNORE_Wdeprecated_declarations \
 						_Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"")
-	#define PRAGMA_DIAGNOSTIC_PUSH		_Pragma("GCC diagnostic push")
-	#define PRAGMA_DIAGNOSTIC_POP		_Pragma("GCC diagnostic pop")
+	#if  __GNUC__ >= 5
+		#define PRAGMA_DIAGNOSTIC_PUSH \
+			_Pragma("GCC diagnostic push")
+		#define PRAGMA_DIAGNOSTIC_POP \
+			_Pragma("GCC diagnostic pop")
+	#else
+		#define PRAGMA_DIAGNOSTIC_PUSH
+		#define PRAGMA_DIAGNOSTIC_POP
+	#endif
 #else
 	#define PRAGMA_INGORE_Wswitch_enum
 	#define PRAGMA_IGNORE_Wsign_compare
-- 
2.21.1

